source/jormungandr/jormungandr/scenarios/helper_classes/fallback_durations.py

```diff
@@ -101,17 +103,22 @@ def __init__(
     def _get_duration(self, resp, place):
         if resp.routing_status == response_pb2.reached:
             return resp.duration
-        return int((place.distance * sqrt(2)) / self._speed_switcher.get(self._mode))
+        return self._get_manhattan_duration(place.distance, self._speed_switcher.get(self._mode))
+
+    def _get_manhattan_duration(self, distance, speed):
+        return int((distance * sqrt(2)) / speed)
 
     @new_relic.distributedEvent("routing_matrix", "street_network")
-    def _get_street_network_routing_matrix(self, origins, destinations):
+    def _get_street_network_routing_matrix(self, origins, destinations, mode=None):
+        if mode == None:
+            mode = self._mode
         with timed_logger(self._logger, 'routing_matrix_calling_external_service'):
             try:
                 return self._streetnetwork_service.get_street_network_routing_matrix(
                     self._instance,
                     origins,
                     destinations,
-                    self._mode,
+                    mode,
```

Comment:
actually, you don't need to change the mode here, we should let the connector(aka streetnetwork calculator) decide which mode it should use. In our case,  whatever the mode is(`car` or `car_no_park`), from now on, kraken should always use  'car_no_park' mode for matrix and direct_path just like other streetnetwork calculators. I know this may be not very clear, let's discuss this later